### DISCUSSION TITLE
Input rightSection should have dynamic width

### DISCUSSION THREAD

**Discussion by moimael** [mantinedev/mantine#6108] [Feature requests]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.8.0

### What package has an issue?

@mantine/core

### What framework do you use?

Other, I will specify in the bug description

### In which browsers you can reproduce the issue?

All

### Describe the bug

The rightSection of the inputs has been a bit of a pain to work with recently, so I'd like to suggest an improvement.

Currently, the rightSection width must be set manually to a fixed size if we want something that doesn't fit in the default width. The value has to be in px, we cannot pass auto or anything similar.

This is an issue when building dynamic forms from an API response.

In our case, we are using the rightSection to display units, which can sometimes be quite long. 

Ex: `$` is fine, but sometimes we have `$ / semester` or `$ / side / year`. The thing is, we cannot know that beforehand, leading to a broken UI. We could try to render and measure the text offscreen before setting the right width but that seems unnecessarily complicated. 
Setting a very wide fixed size to workaround is also a very bad experience because the actual input is pushed way too far to the left in that case (with right aligned inputs).

How about sizing the section dynamically based on content and eventually allow to pass a min and maxWidth?

Thanks!

### If possible, include a link to a codesandbox with a minimal reproduction

[link](https://codesandbox.io/p/sandbox/mantine-react-template-forked-sqfwx8?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsj1t2s500062e666ffbfogt%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsj1t2s500022e66dc26yp4v%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsj1t2s500032e66s47svqfw%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsj1t2s500052e66rr8swv6t%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsj1t2s500022e66dc26yp4v%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsj1t2s400012e662ed1e05k%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpublic%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clsj1tj41004c2e66stt8njp6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A4%252C%2522startColumn%2522%253A32%252C%2522endLineNumber%2522%253A4%252C%2522endColumn%2522%253A32%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clsj1t2s500022e66dc26yp4v%2522%252C%2522activeTabId%2522%253A%2522clsj1t2s400012e662ed1e05k%2522%257D%252C%2522clsj1t2s500052e66rr8swv6t%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsj1t2s500042e66o2j8d7nz%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clsj1t2s500052e66rr8swv6t%2522%252C%2522activeTabId%2522%253A%2522clsj1t2s500042e66o2j8d7nz%2522%257D%252C%2522clsj1t2s500032e66s47svqfw%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clsj1t2s500032e66s47svqfw%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
It is not planned to implement this feature. rightSection has absolute position and changing that will be a huge breaking change.

**Reply by moimael** [reply]
I understand and I agree it would be a breaking change. Could this be considered for the next major version though? 

Another solution could be to allow passing `width: auto`? Currently, we can pass it but it has no effect (it's saved in the CSS var but the way it's used makes auto useless)
